app/lib/server/functions/createDirectRoom.js

```diff
@@ -88,9 +89,9 @@ export const createDirectRoom = function(members, roomExtraData = {}, options =
 			$setOnInsert: generateSubscription(members[0].name || members[0].username, members[0].username, members[0], { ...options.subscriptionExtra }),
 		});
 	} else {
-		members.forEach((member) => {
+		const membersWithPreferences = Users.find({}, { username: 1, 'settings.preferences': 1 });
```

Comment:
you forgot the query to get only the DM members 😱

Reply:
whoops 🙈  I copied the line inside the iteration and forgot to add the query